python/ray/data/iterator.py

```diff
@@ -40,6 +41,7 @@
 if TYPE_CHECKING:
     import tensorflow as tf
     import torch
+    from torch._prims_common import DeviceLikeType
```

Comment:
Similar to the proposed changes in `python/ray/data/dataset.py`. Let's avoid importing private PyTorch APIs, it’s not guaranteed to be stable across PyTorch releases.

Reply:
thx for fixing!Addresses in commit 58a79c12416447294f42723ccf88e1bfea2f3c9f